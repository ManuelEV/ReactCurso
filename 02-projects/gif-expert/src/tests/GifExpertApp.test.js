import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import {GifExpertApp} from '../GifExpertApp';


describe('Pruebas en el componente GifExpertApp', () => {


    test('debe de coincidir el SnapShot', () => {

        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar una lista de categorÃ­as', () => {
        const categories = ['One Punch', 'Shingeki'];
        const wrapper = shallow(<GifExpertApp defaultCategories= {categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });

})