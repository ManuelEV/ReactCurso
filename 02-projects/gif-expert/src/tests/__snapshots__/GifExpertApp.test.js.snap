// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifExpertApp debe de coincidir el SnapShot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en el componente GifExpertApp debe de mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Shingeki"
      key="Shingeki"
    />
  </ol>
</Fragment>
`;
